specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2025-04-01/ResourceHealth.json

```diff
@@ -0,0 +1,2330 @@
+            "$ref": "#/parameters/FilterParameter"
+          },
+          {
+            "$ref": "#/parameters/QueryStartTimeParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The list of current service health events in the subscription",
+            "schema": {
+              "$ref": "#/definitions/events"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ListEventsBySubscriptionId": {
+            "$ref": "./examples/Events_ListBySubscriptionId.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ResourceHealth/events/{eventTrackingId}": {
+      "get": {
+        "tags": [
+          "Event"
+        ],
+        "operationId": "Event_GetBySubscriptionIdAndTrackingId",
+        "description": "Service health event in the subscription by event tracking id",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/FilterParameter"
+          },
+          {
+            "$ref": "#/parameters/QueryStartTimeParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/EventTrackingIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Service Health Event in the subscription requested by tracking id.",
+            "schema": {
+              "$ref": "#/definitions/event"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SecurityAdvisoriesEventBySubscriptionIdAndTrackingId": {
+            "$ref": "./examples/Event_GetBySubscriptionIdAndTrackingId.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ResourceHealth/events/{eventTrackingId}/fetchEventDetails": {
+      "post": {
+        "tags": [
+          "Event"
+        ],
+        "operationId": "Event_fetchDetailsBySubscriptionIdAndTrackingId",
+        "description": "Service health event details in the subscription by event tracking id. This can be used to fetch sensitive properties for Security Advisory events",
```

Comment:
>This can be used to fetch sensitive properties [](http://example.com/codeflow?start=95&length=46)

API deisgn:
should customer need to opt in to fetching sensitive data, or at least be able to opt out or ask it to be redacted, so as not to do it accidentally, in contexts where they don't want TMI? (too much information -- keep it need to know)

Reply:
(imagine you want to support implementing a UX which shows the event with sensitive data masked out first, then someone clicks 'show sensitive details' if needed. The more secure way to implement this is to not request the sensitive data until you actually desire to display it right)